cocos/scene-graph/node.ts

```diff
@@ -1160,7 +1160,7 @@ export class Node extends CCObject implements ISchedulable, CustomSerializable {
      * ```
      */
     public off (type: string, callback?: AnyFunction, target?: unknown, useCapture: any = false): void {
-        this._eventProcessor.off(type, callback, target, useCapture);
+        this._eventProcessor.off(type as NodeEventType, callback, target, useCapture);
```

Comment:
Why not just change the type of `type` from `string` to `NodeEventType `?

Reply:
it's a public interface, the node currently support registering any custom event name, not only NodeEventType

NodeEventType is actually string value but it's some engine internal enumerate event

Reply:
If so, then internally should also be string too.

Reply:
I've created a new issue to deprecate `node.on(type: string)`  
https://github.com/cocos/cocos-engine/issues/15557